server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -44,13 +44,13 @@ public class FieldFetcher {
      */
     private static final int AUTOMATON_MAX_DETERMINIZED_STATES = 100000;
 
-    public static FieldFetcher create(SearchExecutionContext context, Collection<FieldAndFormat> fieldAndFormats) {
+    public static FieldFetcher create(ValueFetchContext context, Collection<FieldAndFormat> fieldAndFormats) {
         Set<String> nestedMappingPaths = context.nestedLookup().getNestedMappers().keySet();
```

Comment:
Is this the only place `nestedLookup()` is used?  I'd be tempted to try and pull it out into FetchContext and then pass NestedLookup as a separate parameter here?